test/upgrade/prober/wathola/sender/services.go

```diff
@@ -30,8 +33,15 @@ import (
 	"time"
 )
 
-var log = config.Log
-var senderConfig = &config.Instance.Sender
+var (
+	// ErrEndpointTypeNotSupported is raised if configured endpoint isn't
+	// supported by any of the event senders that are registered.
+	ErrEndpointTypeNotSupported = errors.New("given endpoint isn't " +
+		"supported by any registered event sender")
+	log          = config.Log
+	senderConfig = &config.Instance.Sender
+	eventSenders = make([]EventSender, 0, 1)
```

Comment:
```suggestion
	eventSenders []EventSender
```

Reply:
This should work as well and is shorter/easier to read.

Reply:
I'm creating a slice with capacity of 1, because I know I'll be adding there at least one element - the default `EventSender`. In that way Golang would not need to grow that slice when appending.

See: https://www.calhoun.io/how-to-use-slice-capacity-and-length-in-go#why-do-we-bother-with-capacity-at-all-if-append-handles-it

Reply:
The fact is that you don't know how many elements you will have in the end. As you're "ranging" over the slice I suppose there can be more than one.
Reserving the capacity is good when you know exactly how many elements you'll have otherwise it's unnecessary micro-optimization at the cost of more complex syntax.
What i'm suggesting is a simpler syntax and alloacting the array later of 1 element later by calling the append itself. The performance is really not different (hardly measurable) but the syntax is easier.
For the test below where you know there will be only one element, it makes more sense to reserve it in advance but that's a minor thing.

Reply:
In production code I know there will be at least one `EventSender`. If user do not provide custom implementation of `EventSender` then the default `httpSender` is used. That's why **I know for sure** there will be at least one element in that slice.

Reply:
OK. I raised my points. I prefer readability over micro-optimization. But nevermind.